dom/fuzzer/modules/web-speech.js

```diff
@@ -0,0 +1,173 @@
+/*
+ * WebSpeech References
+ *
+ * Implementation: https://bugzilla.mozilla.org/show_bug.cgi?id=1003439
+ * WebIDL:
+    dom/webidl/SpeechGrammarList.webidl
+    dom/webidl/SpeechGrammar.webidl
+    dom/webidl/SpeechRecognitionAlternative.webidl
+    dom/webidl/SpeechRecognitionError.webidl
+    dom/webidl/SpeechRecognitionEvent.webidl
+    dom/webidl/SpeechRecognitionResultList.webidl
+    dom/webidl/SpeechRecognitionResult.webidl
+    dom/webidl/SpeechRecognition.webidl
+    dom/webidl/SpeechSynthesisErrorEvent.webidl
+    dom/webidl/SpeechSynthesisEvent.webidl
+    dom/webidl/SpeechSynthesisUtterance.webidl
+    dom/webidl/SpeechSynthesisVoice.webidl
+    dom/webidl/SpeechSynthesis.webidl
+ * Mochitests: dom/media/webspeech/recognition/test
+ *             dom/media/webspeech/synth/test
+ * animation/test/css-animations
+ * Specification: https://dvcs.w3.org/hg/speech-api/raw-file/tip/speechapi.html
+ * MDN: https://developer.mozilla.org/en-US/docs/Web/API/Web_Speech_API
+ *
+**/
+var fuzzerWebSpeech = (function() {
+  var utterances = [];
```

Comment:
It shouldn't be necessary to store a list of [strings that evaluate to] existing utterances. By using `Things.instance("SpeechSynthesisUtterance")` instead of `Random.pick(utterances)`, you gain the ability to operate on objects created by other modules, in particular the evil ones (`WebIDL.js` and `reflection-and-torture.js`). You also save effort by skipping ones that weren't created successfully, or where the code to create them hasn't run yet (e.g. makeCommand was called by a makeSubCommand), or where the entry in `o[]` was deleted in order to test garbage collection.